pkg/util/fs_util.go

```diff
@@ -71,25 +71,58 @@ var excluded []string
 
 type ExtractFunction func(string, *tar.Header, io.Reader) error
 
+type FSConfig struct {
+	includeWhiteout bool
+	extractFunc     ExtractFunction
+}
+
+type FSOpt func(*FSConfig)
+
+func IncludeWhiteout() FSOpt {
+	return func(opts *FSConfig) {
+		opts.includeWhiteout = true
+	}
+}
+
+func ExtractFunc(extractFunc ExtractFunction) FSOpt {
```

Comment:
I couldn't wrap my head around ExtractFunc taking an ExtractFunction type with parameter name extractFunc. I don't have a recommendation either since my go skills are rather poor so just fyi.

Reply:
WDYT about `func ExtractFuncOpt(f ExtractFunction) FSOpt`

Reply:
ah I know get it was well, might just be me not getting it the other night, never seen this pattern before. I think keeping as is is fine